core/src/services/hdfs_native/backend.rs

```diff
@@ -238,10 +249,16 @@ impl Access for HdfsNativeBackend {
         ))
     }
 
-    async fn list(&self, path: &str, _args: OpList) -> Result<(RpList, Self::Lister)> {
+    async fn list(&self, path: &str, args: OpList) -> Result<(RpList, Self::Lister)> {
         let p = build_rooted_abs_path(&self.root, path);
-        let l = HdfsNativeLister::new(p, self.client.clone());
-        Ok((RpList::default(), Some(l)))
+        info!("backend list started. path {} p {}", &path, &p);
+        let iter = self.client.list_status_iter(&p, args.recursive());
+        let stream = iter.into_stream();
```

Comment:
We can store `ListStatusIterator` directly in `Lister` and don't need to play with stream.